cibuildwheel/__main__.py

```diff
@@ -222,8 +231,12 @@ def main() -> None:
     ) or get_requires_python_str(package_dir)
     requires_python = None if requires_python_str is None else SpecifierSet(requires_python_str)
 
+    # Hardcode pre-releases here, current: Python 3.10
     build_selector = BuildSelector(
-        build_config=build_config, skip_config=skip_config, requires_python=requires_python
+        build_config=build_config,
+        skip_config=skip_config,
+        requires_python=requires_python,
+        filter_prerelease="" if prerelease_pythons else "310",
```

Comment:
How about, instead of matching a little bit of the string, we do instead `exclude_prerelease="" if prerelease_pythons else "cp310-*",`, and match that in BuildSelector using wcmatch/fnmatch?

Reply:
That would also allow us to do a pypy release in the future, or a prerelease of a new arch or something.

Reply:
I've reworked this with `skip_prerelease` and expanded the skip_patterns of the build selector with that.

Reply:
@henryiii, @joerick, does the rework work for you ?

Reply:
Hey @mayeut, thanks for doing that! I think this works for me. One side effect of this implementation is that it shows up at the start of the build in the build options printout. What do you think, is this a little confusing for users? 

https://github.com/pypa/cibuildwheel/pull/675/checks?check_run_id=2662524426#step:6:25

I suppose it should show up somewhere in the build options... this printout isn't that intuitive for users anyway, because they set 'build', 'skip', and it's printed here as `build_selector`. And I like how simple this implementation is. So I think this is okay.

Reply:
> One side effect of this implementation is that it shows up at the start of the build in the build options printout. What do you think, is this a little confusing for users?

That's the only thing I don't like with this implementation. As you said, the current printout was already non intuitive so went with that anyway because it keeps the implementation real simple.
We could get the printout to say something like: `build_selector: BuildSelector('*' - '' - 'cp310-*')` but I'm not sure it's better (in fact I'm pretty sure it's worse).

Reply:
I had a go at some refactoring to neaten this up - let me know what you think:

https://github.com/pypa/cibuildwheel/commit/97be46543c6b2fb7baaf67b9eeeb5d02f7b95cca

You can cherry pick it onto this branch if you like.

---

(aside - does Github need a PR-for-a-PR feature? Or should I just push to a PR and say 'feel free to revert'?)

Reply:
You can make a PR to a PR branch. But it's easier (and more visible) if it's in the same repo, since the PR to a PR will show up in the repo it is targeting.

Reply:
Ah, yes, good idea. In doing so I just discovered there's a bug in it actually!

Reply:
ok, so the new commit is 219ead542f5bd54d08ea894db68ab77add151359 🤭

Before:

```
     _ _       _ _   _       _           _
 ___|_| |_ _ _|_| |_| |_ _ _| |_ ___ ___| |
|  _| | . | | | | | . | | | |   | -_| -_| |
|___|_|___|___|_|_|___|_____|_|_|___|___|_|

cibuildwheel version 1.11.0

Build options:
  platform: 'linux'
  architectures: {<Architecture.i686: 'i686'>, <Architecture.x86_64: 'x86_64'>}
  before_all: ''
  before_build: None
  before_test: None
  build_selector: BuildSelector('*' - 'cp310-*')
  build_verbosity: 0

```

After:

```
     _ _       _ _   _       _           _
 ___|_| |_ _ _|_| |_| |_ _ _| |_ ___ ___| |
|  _| | . | | | | | . | | | |   | -_| -_| |
|___|_|___|___|_|_|___|_____|_|_|___|___|_|

cibuildwheel version 1.11.0

Build options:
  platform: 'linux'
  architectures: {<Architecture.i686: 'i686'>, <Architecture.x86_64: 'x86_64'>}
  before_all: ''
  before_build: None
  before_test: None
  build_selector: BuildSelector(build_config='*')
  build_verbosity: 0
```

After with pre-release enabled:

```
     _ _       _ _   _       _           _
 ___|_| |_ _ _|_| |_| |_ _ _| |_ ___ ___| |
|  _| | . | | | | | . | | | |   | -_| -_| |
|___|_|___|___|_|_|___|_____|_|_|___|___|_|

cibuildwheel version 1.11.0

Build options:
  platform: 'linux'
  architectures: {<Architecture.i686: 'i686'>, <Architecture.x86_64: 'x86_64'>}
  before_all: ''
  before_build: None
  before_test: None
  build_selector: BuildSelector(build_config='*', prerelease_pythons=True)
  build_verbosity: 0
```

Reply:
@joerick, This seems much better, will look at integrating this in this branch.